.github/workflows/ci-core.yml

```diff
@@ -1,7 +1,7 @@
 # This workflow will install Python dependencies, run tests and lint with a single version of Python
 # For more information see: https://help.github.com/actions/language-and-framework-guides/using-python-with-github-actions
 
-name: CI - Python application
+name: CI - Python-based Testing
```

Comment:
CI - Python testing?

Reply:
I am thinking of a future where we have a test that actually compiles MLIR, which I would then call C++-based testing. I think Python testing sounds like we are testing Python, whereas the thing I want to say is "test everything that purely runs in Python". What do you think?

Reply:
I would love that, where so you see you that in the timeline? Roughly?

Reply:
I think I can look into getting the CI job running later this week.

My question was more about the name of this CI job (`Python Testing` vs `Python-based Testing`).

Reply:
> I think I can look into getting the CI job running later this week.
That would be awesome!!!
 
> My question was more about the name of this CI job (`Python Testing` vs `Python-based Testing`).
Not a problem for me